// @storybook/angular/dist/builders/start-storybook/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const architect_1 = require("@angular-devkit/architect");
const find_up_1 = require("find-up");
const read_pkg_up_1 = require("read-pkg-up");
const core_common_1 = require("@storybook/core-common");
const telemetry_1 = require("@storybook/telemetry");
const core_server_1 = require("@storybook/core-server");
const run_compodoc_1 = require("../utils/run-compodoc");
const error_handler_1 = require("../utils/error-handler");
const setup_1 = require("../utils/setup");
(0, telemetry_1.addToGlobalContext)('cliVersion', core_common_1.versions.storybook);
const commandBuilder = async (options, context) => {
    const { tsConfig, angularBuilderContext, angularBuilderOptions } = await (0, setup_1.setup)(options, context);
    const docTSConfig = (0, find_up_1.sync)('tsconfig.doc.json', { cwd: options.configDir });
    if (options.compodoc) {
        await (0, run_compodoc_1.runCompodoc)({
            compodocArgs: [...options.compodocArgs, ...(options.quiet ? ['--silent'] : [])],
            tsconfig: docTSConfig ?? tsConfig,
        }, context);
    }
    (0, core_common_1.getEnvConfig)(options, {
        port: 'SBCONFIG_PORT',
        host: 'SBCONFIG_HOSTNAME',
        staticDir: 'SBCONFIG_STATIC_DIR',
        configDir: 'SBCONFIG_CONFIG_DIR',
        ci: 'CI',
    });
    options.port = parseInt(`${options.port}`, 10);
    const { browserTarget, ci, configDir, docs, host, https, port, quiet, enableProdMode = false, smokeTest, sslCa, sslCert, sslKey, disableTelemetry, initialPath, open, debugWebpack, loglevel, webpackStatsJson, statsJson, previewUrl, } = options;
    const standaloneOptions = {
        packageJson: (0, read_pkg_up_1.sync)({ cwd: __dirname }).packageJson,
        ci,
        configDir,
        ...(docs ? { docs } : {}),
        excludeChunks: angularBuilderOptions.styles
            ?.filter((style) => typeof style !== 'string' && style.inject === false)
            .map((s) => s.bundleName),
        host,
        https,
        port,
        quiet,
        enableProdMode,
        smokeTest,
        sslCa,
        sslCert,
        sslKey,
        disableTelemetry,
        angularBrowserTarget: browserTarget,
        angularBuilderContext,
        angularBuilderOptions,
        tsConfig,
        initialPath,
        open,
        debugWebpack,
        webpackStatsJson,
        statsJson,
        loglevel,
        previewUrl,
    };
    const devPort = await runInstance(standaloneOptions);
    return { success: true, info: { port: devPort } };
};
exports.default = (0, architect_1.createBuilder)(commandBuilder);
async function runInstance(options) {
    try {
        const { port } = await (0, core_server_1.withTelemetry)('dev', {
            cliOptions: options,
            presetOptions: { ...options, corePresets: [], overridePresets: [] },
            printError: error_handler_1.printErrorDetails,
        }, () => (0, core_server_1.buildDevStandalone)(options));
        return port;
    }
    catch (error) {
        throw new Error((0, error_handler_1.errorSummary)(error));
    }
}
